File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (w, h) {
    if (w && h > 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    const width = this.width;
    const height = this.height;

    for (let i = 0; i < height; i++) {
      let val = '';
      for (let j = 0; j < width; j++) {
        val += 'X';
      }
      console.log(val);
    }
  }
}

module.exports = Rectangle;

File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
""" Class Rectangle """


class Rectangle:
    """ Initial method """

    def __init__(self, width=0, height=0):
        """ defines a rectangle """

        self.width = width
        self.height = height

    def __str__(self):
        """ print the rectangle """
        if self.__height == 0 or self.__width == 0:
            return ""
        str_new = ('#' * self.__width + '\n') * self.__height
        return str_new.strip('\n')

    @property
    def width(self):
        """using property decorator"""
        return self.__width

    @width.setter
    def width(self, value):
        """Setter and value validation"""
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        else:
            self.__width = value

    @property
    def height(self):
        """using property decorator"""
        return self.__height

    @height.setter
    def height(self, value):
        """Setter and value validation"""
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        else:
            self.__height = value

    def area(self):
        """ Area Rectangle """
        return self.__width * self.__height

    def perimeter(self):
        """ perimeter Rectangle """
        if self.__height == 0 or self.__width == 0:
            return 0
        else:
            p = 2 * (self.__height + self.__width)
            return p
